### PR TITLE
RFC 7232: Directory Service responds with 200 OK when If-Modified-Since

### PR DESCRIPTION
**Pull request by DoruM**
Issue was caused by a difference of less than one second between the value
of the header  If-Modified-Since and the actual date of the modification
of the file.
Additionally the header Last-Modified was not sent when sending the status
304.

I have also took the liberty of closing certain FileInputStream which were
opened but never closed.

### CODE REVIEW

**Comment by cmebarrow** [comment]
@DoruM @sbadugu Please assign this to someone else for review. I am snowed under and will not be available next week since I will be working with a customer.

File: service/http.directory/src/test/java/org/kaazing/gateway/service/http/directory/rfc7232/IfModifiedSinceIT.java [hunk 1]
```diff
@@ -63,14 +63,14 @@
     @Rule
     public final TestRule chain = outerRule(timeout).around(k3po).around(gateway);
 
-    @Ignore("https://github.com/kaazing/gateway/issues/383")
+    //@Ignore("https://github.com/kaazing/gateway/issues/383")
```
Comment (dpwspoon) [inline service/http.directory/src/test/java/org/kaazing/gateway/service/http/directory/rfc7232/IfModifiedSinceIT.java]: Please delete the ignore lines

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpUtils.java [hunk 2]
```diff
@@ -74,6 +74,7 @@
 
 
 	private static final DateFormat[] RFC822_PARSE_PATTERNS = new DateFormat[] {
+	    new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.ENGLISH),
```
Comment (dpwspoon) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpUtils.java]: nit: formatting

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpUtils.java [hunk 3]
```diff
@@ -283,7 +285,8 @@ private static boolean hasBeenModified(File requestFile,
 		}
 
 		// check modified time against file last modified
-		return lastModified > ifModifiedSinceDate.getTime();
+		double time_difference = Math.floor(Math.abs(lastModified - ifModifiedSinceDate.getTime())/1000);
+		return time_difference>0;
```
Comment (dpwspoon) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpUtils.java]: nit: formatting

**Comment by dpwspoon** [comment]
Thanks Dorum, looks good to me and thanks for fixing the resource close.  

Please try to be a little bit tidier with the formatting.  See [this repo](https://github.com/kaazing/code.quality/tree/develop/src/main/resources/org/kaazing/code/quality) for formatters.  On nuklei code based these are strictly enforced.

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpUtils.java [hunk 4]
```diff
@@ -283,7 +285,8 @@ private static boolean hasBeenModified(File requestFile,
 		}
 
 		// check modified time against file last modified
-		return lastModified > ifModifiedSinceDate.getTime();
+		double time_difference = Math.floor(Math.abs(lastModified - ifModifiedSinceDate.getTime())/1000);
```
Comment (dpwspoon) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpUtils.java]: Consider using static imports, I personally don't and sometimes find static import annoying, but our code base seems to be trending in that direction

**Comment by DoruM** [comment]
Done.

**Pull request merged**